lib/resources/personalization.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Personalization resources
+ *
+ * @module resources/order-recipients
+ * @copyright 2015–2023 RewardOps Inc.
+ */
+const axios = require('axios');
+
+const config = require('../config');
+const { setPiiToken } = require('../utils/axios-helpers');
+const { log } = require('../utils/logger');
+
+const piiServerClient = axios.create();
+
+/**
+ *
+ * @param {string} programCode pangea program code
+ * @returns {*} registerMemberTags function
+ */
+function Personalization(programCode) {
+  /**
+   *
+   * @param {*} member consists of foreign_id, segment, member_tags, program_code
+   * @param callback
+   * @returns {string} memberUUID
+   */
+  const registerMemberTags = async (member, callback) => {
```

Comment:
You raise some good points, I know this method mentions that this is for creating or updating a member, but the main reason for this method was to register member tags that we have have on MX with Pangea, and in return Pangea sends us an `id` which is the `memberUUID` for that member, even if we want, we cannot change the name of this method at this point because the corresponding code for this has been merged into the release branch for `MX`, and in order to rename this function we would have to update the `MX` code base as well which is using this method.

Reply:
Ok, just one comment Pangea does not send us id, PII responds member [object](https://github.com/rewardops/pii_service/blob/01b506bd60d5a405f61b5b13b326b08c17e8c780/app/controllers/api/v5/members_controller.rb#L8) (If I'm not mistaken).
Could we postpone this merge a little bit. We call this method on MX side only one time, I could create that technical PR on MX side, and extend existing PR. 

I just need to get approve from you and @benevolentmadman. My opinion is that it's necessary to do it, because otherwise I need to use registerMemberTags for getting member ID in places where I'm going to pass it into Get Items.

Reply:
You don't need to call this method, this method gets called during `login` and `softLogin` flows, that registers the memberUUID with the `userInfo` object, you can access the `userInfo` from the user session, so no need to call this method anymore.

Reply:
@k163791 thanks for clarification.
In this case, my comment is just 'nice to have' and it's perfectly acceptable to set it aside for the time being.